llvm/lib/IR/ProfDataUtils.cpp

```diff
@@ -201,14 +201,23 @@ void scaleProfData(Instruction &I, uint64_t S, uint64_t T) {
                         !ProfDataName->getString().equals("VP")))
     return;
 
+  // If an instruction is a call and its branch weight has more than two
+  // operands, it represents taken vs not-taken branch probabilities and doesn't
+  // need scaling.
+  if (isa<CallBase>(&I) && ProfDataName->getString().equals("branch_weights") &&
```

Comment:
Should we introduce a common API to get the profile annotation type? BR_Weight, Call_Count ...?

Reply:
Probably as a pre-cleanup NFC patch.

Reply:
As discussed added `hasCountTypeMD` in ProfDataUtils.cpp. PTAL, thanks!